src/main/scala/edu/ie3/osmogrid/model/OsmoGridModel.scala

```diff
@@ -93,6 +106,42 @@
     (matchedEntities.par, matchedSubentities)
   }
 
+  def filterForLatestVersions[
+      E <: OsmEntity: ClassTag,
+      S <: OsmEntity: ClassTag
+  ](entities: ParSeq[E], nodes: Map[Long, S]): (ParSeq[E], Map[Long, S]) = {
+
+    val groupedEntities = entities.groupBy(_.id)
+
+    val latestEntities = groupedEntities.values
+      .flatMap { entities =>
+        val maxVersion = entities
+          .flatMap {
+            case entity: E if entity.metaInformation.isDefined =>
+              entity.metaInformation.map(_.version.getOrElse(0))
+            case _ => Some(0)
+          }
+          .reduceOption(_ max _)
+          .getOrElse(0)
+        entities.filter {
+          case entity: E if entity.metaInformation.isDefined =>
+            entity.metaInformation.exists(_.version.contains(maxVersion))
+          case _ => true
+        }
```

Comment:
Just one question from my side. Does this work if we have two versions of the same entity one with metaInformation and one without?
I think in this case, both versions are returned, which could cause problems. Maybe you could add a test for this case.

Reply:
Good point! I changed this into a logger warning and filter out these entities.

Reply:
What happens if we have an entity that occures only once but have no metaInformation? Will it be filtered out and cause a porblem later?

Reply:
As far as I checked there is always a version at the entity. So imho this is very unlikely. But sure we can split the entities and filter only those with multiple occurences. :)

Reply:
@danielfeismann Do you have an update on this part?

Reply:
Honestly, OSM data that includes multiple versions of the same entity is just broken to me. I'd be fine with just assuming that the OSM data we're handling is coherent. Do we really need the meta information then?

Reply:
I agree with you, @sebastian-peter. Tried it again by using overpass, worked well at that location. So I'm fine with closing this one.